risc0/zkvm/src/host/api/convert.rs

```diff
@@ -1206,7 +1209,75 @@ impl TryFrom<pb::api::ProveKeccakRequest> for ProveKeccakRequest {
                 .control_root
                 .ok_or_else(|| malformed_err("ProveKeccakRequest.control_root"))?
                 .try_into()?,
-            input: value.input,
+            input,
         })
     }
 }
+
+pub(crate) fn keccak_input_to_bytes(input: &[KeccakState]) -> Vec<u8> {
+    // Note: safe to cast slice given alignment of KeccakState (8 bytes) is greater than
+    // the alignment of the u8 buffer.
+    bytemuck::cast_slice(input).to_vec()
+}
+
+pub(crate) fn try_keccak_bytes_to_input(input: &[u8]) -> Result<Vec<KeccakState>> {
+    let chunks = input.chunks_exact(std::mem::size_of::<KeccakState>());
+    if !chunks.remainder().is_empty() {
+        bail!("Input length must be a multiple of KeccakState size");
+    }
+    chunks
+        .map(bytemuck::try_pod_read_unaligned)
+        .collect::<Result<_, _>>()
+        .map_err(|e| anyhow!("Failed to convert input bytes to KeccakState: {}", e))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_keccak_bytes_to_input_alignment() {
+        // Create a buffer with extra padding at the start to test different alignments
+        let padding = 8; // Test all possible alignments (0-7)
+        let keccak_states = 3; // Test multiple KeccakStates
+        let state_size = std::mem::size_of::<KeccakState>();
+        let mut test_buffer = vec![0u8; padding + (keccak_states * state_size)];
+
+        // Fill with recognizable pattern
+        for (i, byte) in test_buffer.iter_mut().enumerate() {
+            *byte = (i % 256) as u8;
+        }
+
+        // Test each alignment
+        for offset in 0..8 {
+            let aligned_slice = &test_buffer[offset..][..(keccak_states * state_size)];
+            let result = try_keccak_bytes_to_input(aligned_slice);
+
+            assert!(result.is_ok(), "Failed to parse at alignment {}", offset);
```

Comment:
this can be `{offset}` here and other places below

Reply:
this message doesn't quite make sense. you don't actually know the alignment of the buffer pointer, and then you add some offset to it, which still makes it kinda unknown alignment, maybe just say "offset" here and below instead